### PR TITLE
Add aws-for-fluent-bit

### PR DESCRIPTION
**Pull request by Dentrax**
Source: https://github.com/aws/aws-for-fluent-bit/tree/v2.31.12/scripts/dockerfiles

Fixes:

Related:

#### For new package PRs only

- [ ] This PR is marked as fixing a pre-existing package request bug
  - [ ] Alternatively, the PR is marked as related to a pre-existing package request bug, such as a dependency
- [X] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [X] REQUIRED - The version of the package is still receiving security updates

#### For security-related PRs

- [ ] The security fix is recorded in `advisories` and `secfixes`

#### For version bump PRs

- [ ] The `epoch` field is reset to 0
- [ ] Patch source: _patch source here_

### CODE REVIEW

File: aws-for-fluent-bit.yaml [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+      tag: v${{package.version}}
+
+  - runs: |
+      # GHSA-p782-xgp4-8hr8
+      go get -u golang.org/x/sys@v0.0.0-20220412211240-33da011f77ad
+
+      go mod tidy
+
+  - runs: |
+      mkdir -p "${{targets.destdir}}"/fluent-bit/etc
```
Comment (rawlingsj) [inline aws-for-fluent-bit.yaml:55]: can we switch this to `"${{targets.destdir}}"/etc/fluent-bit` and add a compat subpackage?

File: aws-for-fluent-bit.yaml [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+      mkdir -p "${{targets.destdir}}"/fluent-bit/parsers
+      cp fluent-bit.conf "${{targets.destdir}}"/fluent-bit/etc/
+      cp AWS_FLB_CHERRY_PICKS "${{targets.destdir}}"/AWS_FLB_CHERRY_PICKS
+      cp ./entrypoint.sh "${{targets.destdir}}"/fluent-bit/entrypoint.sh
+      chmod +x "${{targets.destdir}}"/fluent-bit/entrypoint.sh
+
+  - runs: |
+      git clone --depth=1 https://github.com/fluent/fluent-bit-docker-image.git
+      pushd fluent-bit-docker-image
```
Comment (rawlingsj) [inline aws-for-fluent-bit.yaml:64]: CI is failing with `/bin/sh: pushd: not found` maybe just `cd` for now

**Review by rawlingsj — CHANGES_REQUESTED** [review]

File: aws-for-fluent-bit.yaml [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+      cp AWS_FLB_CHERRY_PICKS "${{targets.destdir}}"/etc/fluent-bit/AWS_FLB_CHERRY_PICKS
+      cp ./entrypoint.sh "${{targets.destdir}}"/etc/fluent-bit/entrypoint.sh
+      chmod +x "${{targets.destdir}}"/etc/fluent-bit/entrypoint.sh
+
+  - working-directory: fluent-bit-docker-image
+    pipeline:
+      - uses: git-checkout
+        with:
+          repository: https://github.com/fluent/fluent-bit-docker-image
+          branch: ${FLB_DOCKER_BRANCH}
```
Comment (rawlingsj) [inline aws-for-fluent-bit.yaml:67]:
```suggestion
          branch: ${{vars.FLB_DOCKER_BRANCH}}
```

**Comment by cpanato** [comment]
@Dentrax need a rebase to update the branch

**Review by rawlingsj — DISMISSED** [review]
putting on hold as we need to look at the approach here

File: aws-for-fluent-bit.yaml [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+environment:
+  contents:
+    packages:
+      - build-base
+      - busybox
+      - ca-certificates-bundle
+      - go
+      - fluent-bit
+      - aws-flb-kinesis-compat
```
Comment (kaniini) [inline aws-for-fluent-bit.yaml:17]: Wouldn't these be runtime dependencies?

**Review by kaniini — DISMISSED** [review]

File: aws-for-fluent-bit.yaml [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+
+  - working-directory: fluent-bit-docker-image
+    pipeline:
+      - uses: git-checkout
+        with:
+          repository: https://github.com/fluent/fluent-bit-docker-image
+          branch: ${{vars.FLB_DOCKER_BRANCH}}
+          expected-commit: 98ce3316ea93751ddd33bc319a8f9d177493155b
+      - runs: |
+          cp conf/parsers*.conf "${{targets.destdir}}"/etc/fluent-bit/etc
```
Comment (kaniini) [inline aws-for-fluent-bit.yaml:62]: Are we sure this is right?  We are copying the same files to two different paths?

File: aws-for-fluent-bit.yaml [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+  version: 2.31.12.20230629
+  epoch: 0
+  description: AWS provides a Fluent Bit image with plugins for both CloudWatch Logs and Kinesis Data Firehose.
+  copyright:
+    - license: Apache-2.0
+  dependencies:
+    runtime:
+      - ca-certificates-bundle
+      - fluent-bit
+      - go
```
Comment (kaniini) [inline aws-for-fluent-bit.yaml:12]: Why is `go` a runtime dependency?  That would result in `go` being in the generated image, for example.

Comment (kaniini) [reply aws-for-fluent-bit.yaml:12]: I think we can just drop `go` in general here.

File: aws-for-fluent-bit.yaml (hunk 5 above)
Comment (Dentrax) [reply aws-for-fluent-bit.yaml:62]: Yep, that's weird. But this is what they did: https://github.com/aws/aws-for-fluent-bit/blob/69366b130a28e14c3c9548eabd69b499bb128c7b/scripts/dockerfiles/Dockerfile.build#L61-L64

File: aws-for-fluent-bit.yaml (hunk 6 above)
Comment (Dentrax) [reply aws-for-fluent-bit.yaml:12]: Ahhh I was copied from your review. 🙈 https://github.com/wolfi-dev/os/pull/3473#discussion_r1259931643 Let me remove.

File: aws-for-fluent-bit.yaml (hunk 5 above)
Comment (kaniini) [reply aws-for-fluent-bit.yaml:61]: Ok, just wanted to be sure.

**Review by kaniini — APPROVED** [review]

**Pull request merged**